hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestNSSummaryEndpointWithOBSAndLegacy.java

```diff
@@ -904,6 +975,101 @@ public void testNormalizePathUptoBucket() {
+
+    // creationDate filter passed same as KEY6 and KEY7 creation date, but replicationType filter is EC,
+    // so listKeys API will return zero keys, because no EC key got created at or after creationDate filter value.
+    bucketResponse = nsSummaryEndpoint.listKeysWithDu("EC",
+        "04-04-2024 12:30:00", 0, BUCKET_FIVE_PATH, 10, false);
+    duBucketResponse = (DUResponse) bucketResponse.getEntity();
+    // There are no sub-paths under this OBS bucket.
+    assertEquals(0, duBucketResponse.getCount());
+
+    // creationDate filter passed same as KEY7 creation date, but replicationType filter is RATIS,
+    // so listKeys API will return ZERO keys, as only 1 RATIS key got created at or after creationDate filter value.
+    bucketResponse = nsSummaryEndpoint.listKeysWithDu("RATIS",
+        "04-05-2024 12:30:00", 0, BUCKET_FIVE_PATH, 10, false);
+    duBucketResponse = (DUResponse) bucketResponse.getEntity();
+    // There are no sub-paths under this OBS bucket.
+    assertEquals(1, duBucketResponse.getCount());
+
+    // creationDate filter passed same as KEY6 creation date, and replicationType filter is RATIS,
+    // so listKeys API will return only KEY6, as only one RATIS key got created at or after creationDate filter value,
+    // but since keySize filter value is 110 bytes and all RATIS keys created are of size 100 bytes, so KEY6 will be
+    // filtered out and API will return ZERO keys.
+    bucketResponse = nsSummaryEndpoint.listKeysWithDu("RATIS",
+        "04-04-2024 12:30:00", 110, BUCKET_FIVE_PATH, 10, false);
+    duBucketResponse = (DUResponse) bucketResponse.getEntity();
+    // There are no sub-paths under this OBS bucket.
+    assertEquals(0, duBucketResponse.getCount());
+
+    // creationDate filter passed same as KEY6 creation date, and replicationType filter is EC,
+    // so listKeys API will return only KEY7, as only one EC key got created at or after creationDate filter value,
+    // but since keySize filter value is 110 bytes and all EC keys created are of size 100 bytes, so KEY7 will be
+    // filtered out and API will return ZERO keys.
+    bucketResponse = nsSummaryEndpoint.listKeysWithDu("EC",
+        "04-04-2024 12:30:00", 110, BUCKET_FIVE_PATH, 10, false);
+    duBucketResponse = (DUResponse) bucketResponse.getEntity();
+    // There are no sub-paths under this OBS bucket.
+    assertEquals(0, duBucketResponse.getCount());
+
+    assertEquals(BUCKET_FIVE_DATA_SIZE, duBucketResponse.getSize());
+  }
+
+  @Test
+  public void testListKeysBucketSix() throws Exception {
+    // filter list keys under bucketSix based on RATIS ReplicationConfig and key creation date
+    Response bucketResponse = nsSummaryEndpoint.listKeysWithDu("RATIS",
+        "04-04-2024 12:20:00", 0, BUCKET_SIX_PATH, 10, false);
+    DUResponse duBucketResponse = (DUResponse) bucketResponse.getEntity();
+    // There are no sub-paths under this LEGACY bucket.
+    assertEquals(2, duBucketResponse.getCount());
```

Comment:
We are mentioning that there are no sub paths under this legacy bucket but we are asserting that the `subPathCount` returned by `duBucketResponse.getCount()` returns **2** 
Or does the comment is mentioning something else that I have mistaken?